m2c/arch_mips.py

```diff
@@ -1848,3 +1851,284 @@ def function_return(expr: Expression) -> Dict[Register, Expression]:
+        "fa1": Register("f13"),
+        "fa2": Register("f14"),
+        "fa3": Register("f15"),
+        "fa4": Register("f16"),
+        "fa5": Register("f17"),
+        "fa6": Register("f18"),
+        "fa7": Register("f19"),
+        "fs0": Register("f20"),
+        "ft8": Register("f21"),
+        "fs1": Register("f22"),
+        "ft9": Register("f23"),
+        "fs2": Register("f24"),
+        "ft10": Register("f25"),
+        "fs3": Register("f26"),
+        "ft11": Register("f27"),
+        "fs4": Register("f28"),
+        "ft12": Register("f29"),
+        "fs5": Register("f30"),
+        "ft13": Register("f31"),
+    }
+
+    numeric_regs = {
+        "0": Register("zero"),
+        "1": Register("at"),
+        "2": Register("v0"),
+        "3": Register("v1"),
+        "4": Register("a0"),
+        "5": Register("a1"),
+        "6": Register("a2"),
+        "7": Register("a3"),
+        "8": Register("a4"),
+        "9": Register("a5"),
+        "10": Register("a6"),
+        "11": Register("a7"),
+        "12": Register("t4"),
+        "13": Register("t5"),
+        "14": Register("t6"),
+        "15": Register("t7"),
+        "16": Register("s0"),
+        "17": Register("s1"),
+        "18": Register("s2"),
+        "19": Register("s3"),
+        "20": Register("s4"),
+        "21": Register("s5"),
+        "22": Register("s6"),
+        "23": Register("s7"),
+        "24": Register("t8"),
+        "25": Register("t9"),
+        "26": Register("k0"),
+        "27": Register("k1"),
+        "28": Register("gp"),
+        "29": Register("sp"),
+        "30": Register("fp"),
+        "31": Register("ra"),
+    }
+
+    aliased_regs = {**o32abi_float_regs, **aliased_gp_regs, **numeric_regs}
+
+    def default_function_abi_candidate_slots(self) -> List[AbiArgSlot]:
+        return [
+            AbiArgSlot(0, Register("f12"), Type.floatish()),
+            AbiArgSlot(4, Register("f13"), Type.floatish()),
+            AbiArgSlot(8, Register("f14"), Type.floatish()),
+            AbiArgSlot(12, Register("f15"), Type.floatish()),
+            AbiArgSlot(16, Register("f16"), Type.floatish()),
+            AbiArgSlot(20, Register("f17"), Type.floatish()),
+            AbiArgSlot(24, Register("f18"), Type.floatish()),
+            AbiArgSlot(28, Register("f19"), Type.floatish()),
+            AbiArgSlot(0, Register("a0"), Type.intptr()),
+            AbiArgSlot(4, Register("a1"), Type.any_reg()),
+            AbiArgSlot(8, Register("a2"), Type.any_reg()),
+            AbiArgSlot(12, Register("a3"), Type.any_reg()),
+            AbiArgSlot(16, Register("a4"), Type.any_reg()),
+            AbiArgSlot(20, Register("a5"), Type.any_reg()),
+            AbiArgSlot(24, Register("a6"), Type.any_reg()),
+            AbiArgSlot(28, Register("a7"), Type.any_reg()),
+        ]
+
+    def function_abi(
+        self,
+        fn_sig: FunctionSignature,
+        likely_regs: Dict[Register, bool],
+        *,
+        for_call: bool,
+    ) -> Abi:
```

Comment:
Can you add a comment to this function and to ArchPpc.function_abi that they duplicate each other?